lib/Ama/Model/Votes.pm

```diff
@@ -18,6 +19,19 @@ sub cast {
     }
     $self->pg->db->query($sql, $entry_type, $entry_id, $vote, $self->username, $entry_id, $entry_type, $entry_id, $entry_type, $entry_id)->hash;
   };
+  
+  my $votes = eval {$self->pg->db->query('select votes(?, ?) as votes', $entry_type, $entry_id)->hash->{votes}};
+  if ($votes eq $self->threshold){
+    my $results = eval {
+      if ($entry_type eq 'questions'){
+        $sql = 'delete from questions where question_id = ?'
+      }
+      if ($entry_type eq 'comments'){
+        $sql = 'delete from comments where comment_id = ?'
+      }
+      $self->pg->db->query($sql,  $entry_id)
+    };
+  }
```

Comment:
So this is interesting...  This is the cast action of the votes model, yes?  So you wouldn't necessarily be expecting the cast action to cause a delete.  But that's ok, you can document that...  But how does the action behave when a delete occurs?  It seems it should behave in a way to indicate that a delete operation took place.

Moreover, I think there should be a delete method in the votes table that the cast method would call on.  What if other methods want to be able to delete?  Code re-use...!

For Perl, by convention, methods that should not be called on directly should be named beginning with _

What actions can you do with votes?  You can cast and uncast.  Can you explicitly *delete* a vote?  Maybe...  If yes, have a method name "delete", if no, have a method named "_delete".  Then when other methods in the votes model need to delete a vote, they simply call it with $self->_delete(123);